mesonbuild/interpreter.py

```diff
@@ -2253,6 +2255,21 @@ def get_cross_property_method(self, args, kwargs) -> str:
                 return args[1]
             raise InterpreterException('Unknown cross property: %s.' % propname)
 
+    @noArgsFlattening
+    @permittedKwargs({})
+    @FeatureNew('meson.has_cross_property', '0.58.0')
+    def has_cross_property_method(self, args, kwargs) -> str:
+        if len(args) != 1:
+            raise InterpreterException('Must have exactly one argument.')
+        propname = args[0]
+        if not isinstance(propname, str):
+            raise InterpreterException('Property name must be string.')
+        try:
+            props = self.interpreter.environment.properties.host
+            return propname in props
+        except Exception:
+            return False
```

Comment:
Why try? What specific exception could it raise?

Reply:
Because I blindly cargo-culted/copied it without thinking :)

Fixed now by using Dylan's massively simplified version.